File: src/test/java/com/roscopeco/jasm/model/RefArrayTests.java
package com.roscopeco.jasm.model;

public interface RefArrayTests {
    String[] newSingleElementArray();
    void putInArray(final String[] array, final String data);
    String getFromArray(final String[] array);
    int getArrayLength(final String[] array);

    String[][] multiANewArrayTest();
    String[][] multiANewArrayExplicitDimsTest();
}

File: src/test/kotlin/com/roscopeco/jasm/model/RefArrayTests.kt
package com.roscopeco.jasm.model

interface RefArrayTests {
    fun newSingleElementArray(): Array<String?>
    fun putInArray(array: Array<String?>, data: String)
    fun getFromArray(array: Array<String?>): String
    fun getArrayLength(array: Array<String?>): Int
    fun multiANewArrayTest(): Array<Array<String?>?>
    fun multiANewArrayExplicitDimsTest(): Array<Array<String?>?>
}